cmd/fs-v1.go

```diff
@@ -1042,3 +1058,63 @@ func (fs *fsObjects) ListObjectsHeal(bucket, prefix, marker, delimiter string, m
 func (fs *fsObjects) ListBucketsHeal() ([]BucketInfo, error) {
 	return []BucketInfo{}, errors.Trace(NotImplemented{})
 }
+
+// SetBucketPolicies sets policy on bucket
+func (fs *fsObjects) SetBucketPolicies(bucket string, policy policy.BucketAccessPolicy) error {
+	// Parse validate and save bucket policy.
+	policyBytes, err := json.Marshal(policy)
+	if err != nil {
+		return err
+	}
+	// Acquire a write lock on bucket before modifying its configuration.
+	bucketLock := fs.nsMutex.NewNSLock(bucket, "")
+	if err := bucketLock.GetLock(globalOperationTimeout); err != nil {
+		return err
+	}
+	// Release lock
+	defer bucketLock.Unlock()
+
+	return parseAndPersistBucketPolicy(bucket, policyBytes, fs)
+}
+
+// GetBucketPolicies will get policy on bucket
+func (fs *fsObjects) GetBucketPolicies(bucket string) (policy.BucketAccessPolicy, error) {
+	return readBucketPolicy(bucket, fs)
+}
+
+// DeleteBucketPolicies deletes all policies on bucket
+func (fs *fsObjects) DeleteBucketPolicies(bucket string) error {
+	return persistAndNotifyBucketPolicyChange(bucket, policyChange{
+		true, policy.BucketAccessPolicy{},
+	}, fs)
```

Comment:
You can directly call removeBucketPolicy(). In case of dist-xl you can do S3PeersUpdateBucketPolicy()